test-d/partial-on-undefined-deep.ts

```diff
@@ -49,6 +51,8 @@ expectAssignable<{
 	readonly1?: TestingType['readonly1'];
 	readonly2?: TestingType['readonly2'];
 	tuple?: TestingType['tuple'];
+	indexType: Partial<Pick<TestingType['indexType'], 'propertyB'>> & Omit<TestingType['indexType'], 'propertyB'>;
```

Comment:
It would be better to use expectType

```ts
declare const indexType: PartialOnUndefinedDeep<{[k: string]: string | undefined; propertyA: string; propertyB: string | undefined};>
expectType<{[k: string]?: string | undefined; propertyA: string; propertyB?: string | undefined}>(indexType)
```